### PR TITLE
*SecretName: only append suffix when no values are provided

### PR DESCRIPTION
**Pull request by con5cience**
In the Docs: https://github.com/timescale/timescaledb-kubernetes/blob/master/charts/timescaledb-single/admin-guide.md#alternative-way-of-handling-secrets

> Alternativelly [sic] the chart allows referencing existing Secret objects via secret.credentialsSecretName

In the Code: https://github.com/timescale/timescaledb-kubernetes/blob/master/charts/timescaledb-single/templates/secret-patroni.yaml#L13

```
printf "%s-credentials"
```
😡 

In this PR: Append suffixes to the names of secrets objects only when input values are not provided by the user.

### CODE REVIEW

**Review by paulfantom — CHANGES_REQUESTED** [review]
We are relying on the existence of this suffix in tobs. It might be better to document the requirement of this suffix instead of changing the logic.

The same goes for `-certificate` and `-pgbackrest`. We shouldn't change/document this only for one secret.

**Comment by con5cience** [comment]
> We are relying on the existence of this suffix in tobs. It might be better to document the requirement of this suffix instead of changing the logic.
> 
> The same goes for `-certificate` and `-pgbackrest`. We shouldn't change/document this only for one secret.

Change the reference in `tobs` then? That's whole point, to use the raw value that was passed in and not rely on mutating it or requiring it to be named in a specific fashion in order for it to work properly.

I've pushed another commit to do the same for `-certificate` and `-pgbackrest`.

**Review by paulfantom — APPROVED** [review]

File: charts/timescaledb-single/templates/_helpers.tpl [hunk 1]
```diff
@@ -103,13 +103,13 @@ ${HOME}/.pgbackrest_environment
...
 
 {{- define "secrets_certificate" -}}
-{{ printf "%s-certificate" (include "clusterName" .) }}
+{{ printf "%s" (include "clusterName" .) }}
 {{- end -}}
 
 {{- define "secrets_pgbackrest" -}}
-{{ printf "%s-pgbackrest" (include "clusterName" .) }}
+{{ printf "%s" (include "clusterName" .) }}
 {{- end -}}
```
Comment (paulfantom) [inline charts/timescaledb-single/templates/_helpers.tpl:115]: Wouldn't this try, by default, try to create 3 secrets with the same name and as such cause conflicts?

Comment (con5cience) [reply charts/timescaledb-single/templates/_helpers.tpl]: Good point. I'm working with a fresh install scenario, and the changes required for that are pretty clear.

I've added another commit that (I think) uses the suffixes if there are no inputs. I'm not a Helm templating 🧙 so please sanity check that for me.

**Comment by paulfantom** [comment]
LGTM, you just need to sign the CLA so we can merge it.

**Comment by paulfantom** [comment]
Fixed in #373

**Pull request closed without merging**